planner/update.go

```diff
@@ -64,15 +62,18 @@ func (n *updateNode) Next() (bool, error) {
 			}
 
 			n.currentValue = n.results.Value()
-			docID, err := client.NewDocIDFromString(n.currentValue.GetID())
-			if err != nil {
-				return false, err
+
+			docMap := map[string]any{
+				request.DocIDFieldName: n.currentValue.GetID(),
+			}
+			for k, v := range n.input {
+				docMap[k] = v
 			}
-			patch, err := json.Marshal(n.input)
+			doc, err := client.NewDocFromMap(docMap, n.collection.Schema())
```

Comment:
todo: Now that the col vs schema PR is merged, this probably needs to change to:
```
doc, err := client.NewDocFromMap(docMap, n.collection)
```